packages/workspace/src/browser/workspace-storage-service.ts

```diff
@@ -34,9 +34,8 @@ export class WorkspaceStorageService implements StorageService {
 
     @postConstruct()
     protected init() {
-        this.initialized = this.workspaceService.roots.then(() => {
-            this.prefix = this.getPrefix(this.workspaceService.workspace);
-        });
+        this.initialized = this.workspaceService.roots.then(() => this.updatePrefix());
+        this.workspaceService.onWorkspaceLocationChanged(() => this.updatePrefix());
```

Comment:
Should not it go into `then` to avoid races between event and promise?

Reply:
yep, fixed